scripts/taginfo.js

```diff
@@ -19,27 +52,86 @@ function addNetworkTags(project) {
       let network = entry[0],
         definition = entry[1];
 
-      let description = `Roads carrying routes in this network are marked by shields`;
+      let icon = definition.spriteBlank || definition.norefImage;
+      if (Array.isArray(icon)) {
+        icon = icon[0];
+      }
+
+      let icon_url;
+
+      if (icon == undefined) {
+        let shieldGfx = Shields.generateSpriteCtx({}, `shield\n${network}= `);
+        let def = JSON.stringify(shields[network]);
+
+        if (!shieldGfxMap.has(def)) {
+          shieldGfxMap.set(def, shieldGfxMap.size);
+        }
+
+        let network_filename_id = shieldGfxMap.get(def);
+        let save_filename = `dist/shield-sample/shield_${network_filename_id}.svg`;
+
+        if (!fs.existsSync(save_filename)) {
+          shieldGfx.canvas.saveAsSync(save_filename);
+        }
+        icon_url = `https://zelonewolf.github.io/openstreetmap-americana/shield-sample/shield_${network_filename_id}.svg`;
+      } else {
+        icon_url = `https://raw.githubusercontent.com/ZeLonewolf/openstreetmap-americana/main/icons/${icon}.svg`;
+      }
+
+      let description = `Roads carrying routes in this network are marked by `;
+      if (definition.canvasDrawnBlank) {
+        let shapeDef = definition.canvasDrawnBlank;
+        let prettyShapeName = `${shapeDef.drawFunc}`;
+        let prettyFillColor = getNamedColor(shapeDef.params.fillColor, "white");
+        let prettyStrokeColor = getNamedColor(
+          shapeDef.params.strokeColor,
+          "black"
+        );
+
+        switch (shapeDef.drawFunc) {
+          case "roundedRectangle":
+            if (shapeDef.params.radius == 10) {
+              prettyShapeName = "pill";
+            } else {
+              prettyShapeName = "rectangle";
```

Comment:
Nit: “Rectangular shields” would read a little better than “rectangle-shaped shields”.

Reply:
All set - cleaned up these names to be adjectives.